copilot_proxy/utils/utils.py

```diff
@@ -0,0 +1,7 @@
+"""
+Utils that are stand alone, makes it easier run native tests on an supported platform such as Apple Silicon
+"""
+
+
+def prompt_safety_valve(tokens, max_tokens: int, max_model_len: int) -> list:
+    return tokens[len(tokens) - (max_model_len - max_tokens):]
```

Comment:
I think probably you want to trim the prompt by throwing away the beginning rather than the end here – otherwise the completion generated won't be appropriate for the current cursor position.

Reply:
This trims the beginning right? At least in the tests that I've included it behaves that way.